libglusterfs/src/event-epoll.c

```diff
@@ -70,34 +70,32 @@ static int
 __event_slot_alloc(struct event_pool *event_pool, int fd,
                    int notify_poller_death, struct event_slot_epoll **slot)
 {
-    int i = 0;
     int j = 0;
-    int table_idx;
+    int table_idx = 0;
     int gen;
     struct event_slot_epoll *table = NULL;
 
 retry:
 
-    while (i < EVENT_EPOLL_TABLES) {
-        table = event_pool->ereg[i];
+    while (table_idx < EVENT_EPOLL_TABLES) {
+        table = event_pool->ereg[table_idx];
         if (table) {
-            if (table->slots_used == EVENT_EPOLL_SLOTS)
-                return -1;
-            else
+            if (table->slots_used == EVENT_EPOLL_SLOTS) {
+                table_idx++;
+                continue;
+            } else {
                 break; /* break out of the loop */
+            }
         } else {
-            table = __event_newtable(event_pool, i);
-            if (!table)
+            table = __event_newtable(event_pool, table_idx);
+            if (!table) {
                 return -1;
-            else
-                break; /* break out of the loop */
+            } else {
+                break;
+            }
         }
-
-        i++;
     }
 
-    table_idx = i;
-
     for (j = 0; j < EVENT_EPOLL_SLOTS; j++) {
```

Comment:
I think before checking free slot we need to check table_idx should be less than EVENT_EPOLL_TABLES.

Reply:
```c
if (table_idx < EVENT_EPOLL_TABLES) { 
    for (j = 0 ....)
    /* Rest of the code */
} else {
    // Shouldn't we print something to the logs here?
    // It is highly unlikely that we will not find free slots but for edge cases?
    // I remember that 1024 tables is way too much and we will likely find free slots in one of them
    // Should I just add the if block and skip else part then?
}
```
Thoughts?

Reply:
Yes we can

Reply:
Okay. Got it.